analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/Verifier.cs

```diff
@@ -190,7 +201,23 @@ private IEnumerable<Compilation> CompileRazor()
                     File.Copy(file, Path.Combine(tempPath, Path.GetFileName(file)));
                 }
 
-                return new[] { workspace.OpenProjectAsync(Path.Combine(tempPath, "EmptyProject.csproj")).Result.GetCompilationAsync().Result };
+                var path = Path.Combine(tempPath, "EmptyProject.csproj");
+                var xml = XElement.Load(path);
+
+                // Set TargetFramework
+                xml.Descendants("TargetFramework").Single().Value = builder.Framework;
+
+                var langVersion = xml.Descendants("LangVersion").Single();
+                foreach (var parseOption in GetParseOptions())
```

Comment:
In theory, we should not face any problems here, and at the same time just to be on the safe side since the compilation will add folders to the `tempPath `. Let's create a unique folder for each build and put the project there and build it there.